### PR TITLE
apiserver: simplify deepcopy calls

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

**Comment by sttts** [comment]
Splitting up https://github.com/kubernetes/kubernetes/pull/48765 for easier review. Taking over the lgtm from there as these parts haven't changed at all in the rebases.

File: staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go [hunk 1]
```diff
@@ -1071,10 +1059,15 @@ func copyListOptions(options *metainternalversion.ListOptions) *metainternalvers
 // possibly with storage API, but watch is not delivered correctly then).
 // It will be possible to fix it with v3 etcd API.
 func (e *Store) DeleteCollection(ctx genericapirequest.Context, options *metav1.DeleteOptions, listOptions *metainternalversion.ListOptions) (runtime.Object, error) {
+	if listOptions == nil {
+		listOptions = &metainternalversion.ListOptions{}
```
Comment (deads2k) [inline staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go]: pre-existing, but this looks weird, right?

File: staging/src/k8s.io/apiserver/pkg/storage/cacher.go [hunk 2]
```diff
@@ -877,26 +874,11 @@ func (c *cacheWatcher) sendWatchCacheEvent(event *watchCacheEvent) {
 	var watchEvent watch.Event
 	switch {
 	case curObjPasses && !oldObjPasses:
-		object, err := c.copier.Copy(event.Object)
```
Comment (deads2k) [inline staging/src/k8s.io/apiserver/pkg/storage/cacher.go]: is this copier still used somewhere?

Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/storage/cacher.go]: this goes away in a follow-up. These PRs are only the copy/deepcopy calls.

File: staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go (hunk 1 above)
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go]: not necessarily. It means that listOptions is never nil. And indeed line 1071 looks like it needs this.

**Pull request merged**